### PR TITLE
make changes to internal deck deployment for private repositories

### PR DESCRIPTION
**Pull request by droslean**
/cc @stevekuznetsov

### CODE REVIEW

File: core-services/prow/03_deployment/deck.yaml [hunk 1]
```diff
@@ -93,6 +93,7 @@ objects:
           - --job-config-path=/etc/job-config
           - --spyglass=true
           - --rerun-creates-job=true
+          - --gcs-credentials-file=/etc/gce-sa-credentials-gcs-publisher/service-account.json
```
Comment (stevekuznetsov) [inline core-services/prow/03_deployment/deck.yaml]: GCS publisher account should _not_ be the one that has access to the private bucket. `gcs-publisher` is for publishing (not reading) for `origin-ci-test`.

**Comment by droslean** [comment]
/hold

Hold, until we create a new service account.

**Comment by petr-muller** [comment]
> Don't we want this on a private deck instance?

We do.

File: core-services/prow/03_deployment/deck.yaml [hunk 2]
```diff
@@ -284,11 +284,16 @@ objects:
           - -cookie-secret-file=/etc/proxy/secrets/session_secret
           - -tls-cert=/etc/tls/private/tls.crt
           - -tls-key=/etc/tls/private/tls.key
+          - --hidden-only=true
+          - --gcs-credentials-file=/etc/gce-sa-credentials-gcs-private/service-account.json
```
Comment (petr-muller) [inline core-services/prow/03_deployment/deck.yaml]: This does not seem correct - these are args for the `oauth-proxy`, not for deck.

File: core-services/prow/03_deployment/deck.yaml [hunk 3]
```diff
@@ -284,11 +284,16 @@ objects:
...
           volumeMounts:
           - mountPath: /etc/tls/private
             name: deck-internal-tls
           - mountPath: /etc/proxy/secrets
             name: session-secret
+          - name: gce-sa-credentials-gcs-private
+            mountPath: /etc/gce-sa-credentials-gcs-private
+            readOnly: true
```
Comment (petr-muller) [inline core-services/prow/03_deployment/deck.yaml]: Same here, I think oauth-proxy does not need this, deck does.

**Comment by droslean** [comment]
/hold

Further configuration are needed for the `oauth-proxy` container to be able to authenticate against `Openshift-priv` organization.

File: core-services/prow/03_deployment/deck.yaml [hunk 4]
```diff
@@ -297,10 +297,15 @@ objects:
           - --hook-url=http://hook:8888/plugin-help
           - --hidden-only=true
           - --cookie-secret=/etc/proxy/secrets/session_secret
+          - --hidden-only=true
```
Comment (petr-muller) [inline core-services/prow/03_deployment/deck.yaml]: heh, it's already there ;)

Comment (droslean) [reply core-services/prow/03_deployment/deck.yaml]: oh jesus

**Pull request merged**